From 025ac2e860b5d3d53f4c9fe367819c2730e93ed1 Mon Sep 17 00:00:00 2001
From: Sergio Alves <sergio.dasilvalves@arm.com>
Date: Mon, 24 Jul 2023 08:24:10 +0000
Subject: [PATCH] tc2: fix Android with GPU Hw rendering boot on fpga

Signed-off-by: Sergio Alves <sergio.dasilvalves@arm.com>
---
 fdts/tc_fpga.dts | 26 +++++++++++++++++++-------
 1 file changed, 19 insertions(+), 7 deletions(-)

diff --git a/fdts/tc_fpga.dts b/fdts/tc_fpga.dts
index b5f759b5f..014b5f1b3 100644
--- a/fdts/tc_fpga.dts
+++ b/fdts/tc_fpga.dts
@@ -5,9 +5,10 @@
  */
 
 /dts-v1/;
-
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/interrupt-controller/irq.h>
 / {
-	compatible = "arm,tc0";
+	compatible = "arm,tc";
 	interrupt-parent = <&gic>;
 	#address-cells = <2>;
 	#size-cells = <2>;
@@ -497,10 +498,14 @@
 		clock-output-names = "pxclk";
 	};
 
-	smmu_700: smmu_700@3f000000 {
+	mmu_600_tcu: mmu_600_tcu@2ce00000 {
 		compatible = "arm,smmu-v3";
-		reg = <0x0 0x3f000000 0x0 0x5000000>;
-		dma-coherent;
+		reg = <0 0x2ce00000 0 0x100000>;
+		interrupts = <GIC_SPI 75 IRQ_TYPE_EDGE_RISING>,
+			     <GIC_SPI 76 IRQ_TYPE_EDGE_RISING>,
+			     <GIC_SPI 77 IRQ_TYPE_EDGE_RISING>;
+		interrupt-names = "eventq", "cmdq-sync", "gerror";
+		//dma-coherent;
 		#iommu-cells = <1>;
 	};
 
@@ -512,7 +517,7 @@
 		interrupts = <0 69 4>;
 		interrupt-names = "DPU";
 		clocks = <&dpu_aclk>;
-		iommus = <&smmu_700 0x100>;
+		iommus = <&mmu_600_tcu 0>;
 
 		clock-names = "aclk";
 		pl0: pipeline@0 {
@@ -592,11 +597,18 @@
 		clocks = <&gpu_clk>, <&gpu_core_clk>;
 		clock-names = "clk_mali", "shadercores";
 		power_policy = "always_on";
-		iommus = <&smmu_700 0x200>;
 		kvm,protected-device;
 		operating-points = <
 			/* KHz uV */
 			50000 820000
 		>;
+
+		gpu_power_model: power_model {
+			compatible = "arm,mali-simple-power-model";
+			static-coefficient = <2427750>;
+			dynamic-coefficient = <4687>;
+			ts = <20000 2000 (-20) 2>;
+			thermal-zone = "";
+		};
 	};
 };
-- 
2.25.1

