From cef80b29526ed68e7aa6fea25af3d2f6f4a90b68 Mon Sep 17 00:00:00 2001
From: Usama Arif <usama.arif@arm.com>
Date: Tue, 12 Oct 2021 10:29:24 +0100
Subject: [PATCH] driver/mmc: add FPGA variant for mmc

Signed-off-by: Usama Arif <usama.arif@arm.com>
---
 drivers/mmc/host/mmci.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/mmc/host/mmci.c b/drivers/mmc/host/mmci.c
index 3765e2f4ad98..46eb28f12a5f 100644
--- a/drivers/mmc/host/mmci.c
+++ b/drivers/mmc/host/mmci.c
@@ -2385,6 +2385,11 @@ static const struct amba_id mmci_ids[] = {
 		.mask	= 0xff0fffff,
 		.data	= &variant_arm_extended_fifo_hwfc,
 	},
+	{
+		.id	= 0x03041180,
+		.mask	= 0xff0fffff,
+		.data	= &variant_arm_extended_fifo_hwfc,
+	},
 	{
 		.id	= 0x00041181,
 		.mask	= 0x000fffff,
-- 
2.34.1

